pkg/kubectl/genericclioptions/resource/visitor.go

```diff
@@ -376,12 +376,28 @@ func (v FlattenListVisitor) Visit(fn VisitorFunc) error {
 		if info.Object == nil {
 			return fn(info, nil)
 		}
-		items, err := meta.ExtractList(info.Object)
-		if err != nil {
+		if !meta.IsListType(info.Object) {
 			return fn(info, nil)
 		}
-		if errs := runtime.DecodeList(items, v.mapper.decoder); len(errs) > 0 {
-			return utilerrors.NewAggregate(errs)
+
+		items := []runtime.Object{}
+		itemsToProcess := []runtime.Object{info.Object}
```

Comment:
the length of this does not seem to change within the for loop below. Is a slice needed here?

Reply:
I think line 400 is supposed to be appending to `itemsToProcess`

Reply:
probably worth a test

Reply:
> worth a test

added




> I think line 400 is supposed to be appending to itemsToProcess

fixed